### PR TITLE
Add ReturnTypeWillChange to FrameCollection

### PR DESCRIPTION
**Pull request by kylekatarnls**
Fix PHP 8.1 compatibility

### CODE REVIEW

File: src/Whoops/Exception/FrameCollection.php [hunk 1]
```diff
@@ -164,11 +172,22 @@ public function serialize()
...
 
+    public function __serialize(): array
+    {
+        return $this->frames;
+    }
+
+    public function __unserialize(array $serializedFrames): void
+    {
+        $this->frames = $serializedFrames;
+    }
```
Comment (denis-sokolov) [inline src/Whoops/Exception/FrameCollection.php:189]: Are these related to the ReturnTypeWillChange or is this an accidental addition?

**Review by denis-sokolov — COMMENTED** [review]
Thank you, @kylekatarnls! Indeed, we want something like that.

File: src/Whoops/Exception/FrameCollection.php (hunk 1 above)
Comment (kylekatarnls) [reply src/Whoops/Exception/FrameCollection.php:189]: This is also needed for PHP 8.1 compatibility because `Serializable` is deprecated.

Comment (denis-sokolov) [reply src/Whoops/Exception/FrameCollection.php:189]: Thanks!

**Comment by SvenRtbg** [comment]
Guys, this merge destroys PHP 5 compatibility because of the return types.

**Comment by kylekatarnls** [comment]
GitHub Actions started from PHP 7.0, I didn't notice the composer.json still support PHP 5.

If PHP 5 is still meant to be supported, the it should be added to GitHub Actions to be tested.

But maybe it's a mistake in the composer.json. Actually I don't think it will be possible to be compatible with both PHP 5 and 8.1 without deprecation/warning notices in the same branch.

And I know it's not what you want to hear but PHP 8 is out now, so it's really time to drop PHP 5.

If you decide not to upgrade PHP, then the more consistent move could be also to lock the `filp/whoops` version to something like `~2.13.0`

**Comment by SvenRtbg** [comment]
I agree that the existing support for PHP 5, even though untested, is surprising, but it is what is announced in the composer.json.

As you can see from #504 it has been discussed long ago to drop PHP 5 support, but somehow never happened.

**Comment by AmraniCh** [comment]
I think whoops must be able to run in all PHP versions (from PHP 5), PHP 5 is still there in the market and a lot of websites still use it specially in development environment (including me).

**Comment by staabm** [comment]
just have a look at the master branch. denis already added testcoverage https://github.com/filp/whoops/commit/7bf80a3773ca927e3d55db2c07d72e55e295c9a9

**Comment by kylekatarnls** [comment]
> PHP 5 is still there in the market and a lot of websites still use it specially in development environment (including me).

"A lot" is probably exaggerated:
https://blog.packagist.com/php-versions-stats-2021-1-edition/

It's completely negligible in proportion. And if you don't feel using PHP 5 is a problem, you should feel OK to not be up to date with the latest filp/whoops neither 🤷

**Pull request merged**